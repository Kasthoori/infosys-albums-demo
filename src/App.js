import React, { useState, useEffect } from "react";
import axios from "axios";
import { useHistory } from "react-router-dom";
import "./App.css";
import ListItem from "./componenets/ListItem/ListItem";


const App = () => {
  const [albums, setAlbums] = useState([]);
  const [tracks, setTracks] = useState([]);
  const history = useHistory();

  const url1 = "http://jsonplaceholder.typicode.com/albums";
  const url2 = "http://jsonplaceholder.typicode.com/photos";

  document.title = 'Albums Available';

  useEffect(() => {
    async function fetchData() {
      const resp_albums = await axios.get(url1);
      const resp_tracks = await axios.get(url2);

      setAlbums(resp_albums.data);
      setTracks(resp_tracks.data);
    }

    fetchData();
  }, []);

  const listOnClick = (id, title, userId) => {
    let tracks_in_album = tracks.filter((item) => {
      return item.albumId === id;
    });

    history.push("/details", {
      status: "true",
      id: id,
      title: title,
      userId: userId,
      tracks: tracks_in_album,
    });
  };

  if (albums.length === 0) {
    return (
      <div className="App">
        <span data-testid="loading">
          <h4>Loading.....</h4>
        </span>
      </div>
    );
  } else {
    return (
      <div className="App">
        <h1>{document.title}</h1>

        <div className="List">
          <span data-testid="resolved">
            {albums.map((item) => (
              <ListItem
                key={item.id + item.userId}
                id={item.id}
                title={item.title}
                userId={item.userId}
                onClick={listOnClick}
              />
            ))}
          </span>
        </div>
      </div>
    );
  }
};

export default App;
