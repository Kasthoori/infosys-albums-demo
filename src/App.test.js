/* eslint-disable no-empty-pattern */
/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import { cleanup, render, waitForElement } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import toJson from "enzyme-to-json";

import App from "./App";

describe("App Testing", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  afterEach(cleanup);

  it("should check url1", async () => {
    const url1 = "/albums";
    const { getByTestId } = render(<App url={url1} />);
    expect(getByTestId("loading")).toHaveTextContent("Loading.....");

    jest.fn().mockImplementationOnce({ data: { albums: "a6natus" } });
    const resolvedSpan = await waitForElement(() => getByTestId("resolved"));
    expect(resolvedSpan).toHaveTextContent("a6natus");
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render a <span />", () => {
    expect(wrapper.find("span").length).toEqual(1);
  });

  it("renders without crashing", () => {
    const props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: "reactjs",
      posts: [],
    };

    const cont = shallow(<App {...props} />);
    expect(toJson(cont)).toMatchSnapshot();
  });
});
