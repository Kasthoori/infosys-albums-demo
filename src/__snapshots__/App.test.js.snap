// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing renders without crashing 1`] = `
<div
  className="App"
>
  <span
    data-testid="loading"
  >
    <h4>
      Loading.....
    </h4>
  </span>
</div>
`;

exports[`App Testing should render correctly 1`] = `ShallowWrapper {}`;
